src/controllers/bulk/bulkcontroller.cpp

```diff
@@ -175,11 +175,14 @@ int BulkController::open() {
         qCDebug(m_logBase) << "unable to automatically detach kernel driver for" << getName();
     }
 
-    if (int ret = libusb_claim_interface(m_phandle, m_interfaceNumber); ret < 0) {
-        qCWarning(m_logBase) << "Cannot claim interface for" << getName()
-                             << ":" << libusb_error_name(ret);
-        libusb_close(m_phandle);
-        return -1;
+    if (m_interfaceNumber.has_value()) {
+        int error = libusb_claim_interface(m_phandle, *m_interfaceNumber);
+        if (error < 0) {
+            qCWarning(m_logBase) << "Cannot claim interface for" << getName()
+                                 << ":" << libusb_error_name(error);
+            libusb_close(m_phandle);
+            return -1;
+        }
     } else {
```

Comment:
This needs to be moved to the inner scope

Reply:
whoops of course. fixed